File: BackEnd/CoCanvas.Domain/Entities/Post.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CoCanvas.Domain.Entities
{
    public class Post
    {
        public Guid Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;

        // Image storage (either file path, URL, or blob reference)
        public string ImageUrl { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; }
        public string UserId { get; set; } = string.Empty;
        public User User { get; set; } = null!;
        public ICollection<Critique> Critiques { get; set; } = new List<Critique>();
        public ICollection<Tag> Tags { get; set; } = new List<Tag>();
    }
}

File: FrontEnd/src/DTO/post.ts
import type { CritiqueDto } from '@/DTO/critique'

export interface PostDto {
  id: string
  userId: string
  title: string
  rating: number
  description: string
  imageUrl: string
  critiques: CritiqueDto[]
  createdAt: string
  tags: string[]
}

export interface CreatePostDto {
  title: string
  description: string
  userId: string
  tags: string[]
}

export interface DisplayPostDto {
  postId: string
  postTitle: string
  imageUrl: string
}
